src/playlist.js

```diff
@@ -305,6 +307,43 @@ const calculateExpiredTime = function(playlist, expiredSync, segmentSync) {
 
     return segmentSync.time - sumDurations(playlist, syncIndex, 0);
   }
+  return 0;
+};
+
+const calculatePlaylistEnd_ = function(playlist, useSafeLiveEnd) {
+  if (!playlist || !playlist.segments) {
+    return null;
+  }
+  if (playlist.endList) {
+    return duration(playlist);
+  }
+  let { expiredSync, segmentSync } = getPlaylistSyncPoints(playlist);
```

Comment:
Since you moved `getPlaylistSyncPoints` to `calculateExpiredTime` calling it again here just to check for null on `expiredSync` `segmentSync` is unnecessary work. If `calculateExpiredTime` starts returning null when it can't calculate expired, you can just check for `null` on `expired` instead of checking the sync points